modules/consensus/consensusset.go

```diff
@@ -51,16 +42,25 @@ type ConsensusSet struct {
 	// known to the expensive part of block validation.
 	dosBlocks map[types.BlockID]struct{}
 
-	// TODO: add a persistDir
-	mu *sync.RWMutex
+	// checkingConsistency is a bool indicating whether or not a consistency
+	// check is in progress. The consistency check logic call itself, resulting
+	// in infinite loops. This bool prevents that while still allowing for full
+	// granularity consistency checks. Previously, consistency checks were only
+	// performed after a full reorg, but now they are performed after every
+	// block.
+	checkingConsistency bool
```

Comment:
maybe, otoh I'd really rather perform the full suite of consistency checks after every atomic action (adding or removing a block)